diff --git a/dirs/dirs.go b/dirs/dirs.go
index 1569658dc3..f54dac60c6 100644
--- a/src/github.com/snapcore/snapd/dirs/dirs.go
+++ b/src/github.com/snapcore/snapd/dirs/dirs.go
@@ -419,7 +419,7 @@ func SetRootDir(rootdir string) {
 		// both RHEL and CentOS list "fedora" in ID_LIKE
 		DistroLibExecDir = filepath.Join(rootdir, "/usr/libexec/snapd")
 	} else {
-		DistroLibExecDir = filepath.Join(rootdir, "/usr/lib/snapd")
+		DistroLibExecDir = filepath.Join(rootdir, "/usr/lib64/snapd")
 	}
 
 	XdgRuntimeDirBase = filepath.Join(rootdir, "/run/user")
